File: frontend/src/lib/api/companycam.ts
import { apiGet } from "@/lib/api";

export interface CompanyCamPhoto {
  id: string;
  thumbnail_url: string;
  web_url: string;
  captured_at?: number | null;
  creator_name?: string | null;
}

export interface CompanyCamProjectPhotos {
  project_id: string;
  project_name: string;
  project_url: string;
  photo_count: number;
  address?: string | null;
  photos: CompanyCamPhoto[];
}

export interface ContactCompanyCamPhotosResponse {
  connected: boolean;
  projects: CompanyCamProjectPhotos[];
}

export function getContactCompanyCamPhotos(
  workspaceId: string,
  contactId: number
): Promise<ContactCompanyCamPhotosResponse> {
  return apiGet<ContactCompanyCamPhotosResponse>(
    `/api/v1/workspaces/${workspaceId}/contacts/${contactId}/companycam-photos`
  );
}

File: backend/app/schemas/companycam.py
"""CompanyCam contact-photos schemas."""

from pydantic import BaseModel


class CompanyCamPhoto(BaseModel):
    """One photo, reduced to what the sidebar gallery renders."""

    id: str
    thumbnail_url: str
    web_url: str
    captured_at: int | None = None
    creator_name: str | None = None


class CompanyCamProjectPhotos(BaseModel):
    """A matched CompanyCam project and its recent photos."""

    project_id: str
    project_name: str
    project_url: str
    photo_count: int
    address: str | None = None
    photos: list[CompanyCamPhoto]


class ContactCompanyCamPhotosResponse(BaseModel):
    """All CompanyCam projects matched to one CRM contact."""

    connected: bool
    projects: list[CompanyCamProjectPhotos]
